notebook/gateway/managers.py

```diff
@@ -366,24 +411,32 @@ def start_kernel(self, kernel_id=None, path=None, **kwargs):
                         or k in GatewayClient.instance().env_whitelist.split(",")}
 
             # Convey the full path to where this notebook file is located.
-            if path is not None and kernel_env.get('KERNEL_WORKING_DIR') is None:
+            if path and kernel_env.get('KERNEL_WORKING_DIR') is None:
                 kernel_env['KERNEL_WORKING_DIR'] = kwargs['cwd']
```

Comment:
don't think key addition is required to the kwargs at line 398. Here, it can be assigned to `self.cwd_for_path()`

Reply:
again as I said not my code not changing it to something else.